packages/front-generator/src/generators/react-native/app/template/components/Login.tsx

```diff
@@ -0,0 +1,183 @@
+
+  @observable login = '';
+  @observable password = '';
+  @observable loading = false;
+
+  @observable badCredentialsError = false;
+  @observable serverError = false;
+
+  render() {
+    return (
+      <KeyboardAvoidingView style={styles.container} behavior='padding'>
+          <View style={styles.header}>
+            <Text style={styles.header_title}>scr</Text>
+            <Text style={styles.header_subTitle}>Sign in to your account</Text>
+          </View>
+          <View style={styles.form}>
+            {this.badCredentialsError && (
+              <Text style={styles.form_globalError}>
+                Login failed. Unknown login or bad password.
+              </Text>
+            )}
+            {this.serverError && (
+              <Text style={styles.form_globalError}>Server error</Text>
+            )}
+            <TextInput style={[
+                styles.form_input,
+                this.badCredentialsError && styles.form_input__invalid
+              ]}
+              value={this.login}
+              placeholder="Login"
+              placeholderTextColor={colors.placeholders}
+              onChangeText={this.onLoginChange}
+              onSubmitEditing={this.onSubmit}
+            />
+            <TextInput style={[
+                styles.form_input,
+                this.badCredentialsError && styles.form_input__invalid
+              ]}
+              value={this.password}
+              placeholder="Password"
+              placeholderTextColor={colors.placeholders}
+              secureTextEntry={true}
+              onChangeText={this.onPasswordChange}
+              onSubmitEditing={this.onSubmit}
+            />
+            <PrimaryButton onPress={this.onSubmit}
+                           loading={this.loading}
+                           style={styles.form_submitButton}
+                           disabled={!this.isSubmitEnabled()}
+            >
+              Log in
+            </PrimaryButton>
+          </View>
+      </KeyboardAvoidingView>
+    );
+  }
+
+  onLoginChange = (login: string) => {
+    this.badCredentialsError = false;
+    this.login = login;
+  };
+
+  onPasswordChange = (password: string) => {
+    this.badCredentialsError = false;
+    this.password = password;
+  };
+
+  clearErrors = () => {
+    this.badCredentialsError = false;
+    this.serverError = false;
+  };
+
+  isSubmitEnabled = (): boolean => {
+    return this.login.length > 0 && this.password.length > 0;
+  };
+
+  onSubmit = () => {
+    if (!this.isSubmitEnabled) {
+      return;
+    }
+
+    this.clearErrors();
+    this.loading = true;
+
+    this.props
+      .onLoginSubmit(this.login, this.password)
+      .then(_value => {
+        return AsyncStorage.setItem(REST_TOKEN_STORAGE_KEY, cubaREST.restApiToken);
+      })
+      .catch(err => {
+        if (err.response) {
+          err.response.json().then((message) => {
+            if (message.error === 'invalid_grant') {
```

Comment:
I think we need separate file with error constants, with js doc description for each one.